### ISSUE TITLE
Use markdown languages to distinguish blocks

### ISSUE THREAD

**Issue by hayd** [MichaelHatherly/Documenter.jl#5]
This might be a cleaner solution to using {...} in the first line.

IMO it also has the benefit of being more "honest". In the sense that:

    julia> print(1)
    1

is not julia. That said these could *render* to a julia code block...

**Comment by MichaelHatherly** [comment]
For `{docs}`, `{meta}`, etc. ? I thought of doing that and it wouldn't be much trouble to change. My reasoning was that it's easier to tell, when looking at an unrendered markdown file what a block is for if you can directly see that name. I'm open to changing it though.

**Comment by hayd** [comment]
> when looking at an unrendered markdown file

Do you mean when looking at the file *rendered* e.g. in github?

When unrendered the markdown language is easy to see!

    ```repl
    julia> ...
    ```

and

    ```doctest
    julia> ...
    ...
    ```

seem super easy to read AND easy on the eyes when rendered. Also, it's markdown and not a DSL - which seems elegant!

*If you mean it's visible when rendered, I'm a little dubious we should care about that... as then the natural further thing to do is not to even enforce the julia language (as that is likewise hidden).*

**Comment by MichaelHatherly** [comment]
> Do you mean when looking at the file rendered e.g. in github?

Yeah, just when rendered like that. It's quite minor though, so moving to using `.language` for everything is fine with me.

**Comment by hayd** [comment]
This doesn't make sense for content & index, although I think they could use meta:

    ```meta
    content
    ```

**Comment by MichaelHatherly** [comment]
`{contents}` doesn't really fit the meaning of the `{meta}` blocks, which are meant for key/value pairs that get used later in the page. Perhaps the name needs a change, maybe `{config`} or `{settings}`.

Maybe the settings for a particular `{index}`/`{contents}` should appear in the block itself.

```
    {index}
    Modules = [...]
```

or

```
    ```index
    Modules = [...]
    ```
```

I still kind of prefer the `{index}` style to using the language field since it's *perhaps* easier to notice when skimming a template file. (Might just be me though.) It's quite a minor change needed I think so might be worth exploring how it looks.

**Comment by MichaelHatherly** [comment]
Coming back to this after a short break with fresh eyes I'm a bit more amenable to using `.language` for specifying these rather than inline syntax. (I was just bitten by the run-on problem of indented blocks.)

Perhaps the following would be suitable. Use the code block language in place of `{...}` syntax but include some kind of symbol to differentiate from standard markdown code block languages.

Something like

    ```@docs
    ...
    ```

or some other symbol, `#docs`, `!docs`, etc. rather than

    ```docs
    ...
    ```

This would require some churn in packages, but could be deprecated for a little while things get updated. Probably best to do this before `0.1`.

**Comment by hayd** [comment]
`@doc` would make sense, would be interesting to see if text editors barf at that.  `jl @doc` is probably too verbose!

> An info string can be provided after the opening code fence. Opening and closing spaces will be stripped, and the first word, prefixed with language-, is used as the value for the class attribute of the code element within the enclosing pre element.

**Comment by MichaelHatherly** [comment]
Atom, Vim, and gists all support this fine it would seem, I'm sure there won't be much of a problem.

Agreed that `jl @doc` is too verbose, `@docs` should be fine I think. (Plural rather than singular, since it can take multiple "arguments".)

**Comment by MichaelHatherly** [comment]
Closed by #51.

**Comment by MichaelHatherly** [comment]
Thanks for advocating for this change @hayd. Seems to have been a positive reaction to the new syntax from what I can tell in the opened PRs. (I quite like it too.)
